lib/install.js

```diff
@@ -216,9 +217,46 @@ function Installer (where, dryrun, args) {
 }
 Installer.prototype = {}
 
-Installer.prototype.run = function (cb) {
+function updateStatsFile (itWorked) {
+  // If the stats reporter hasn't managed to PUT yet then kill it so that it doesn't
+  // step on our updating the cli-stats json
+  if (npm.statsProcess) npm.statsProcess.kill('SIGKILL')
+  var statsFile = path.join(npm.config.get('cache'), 'anonymous-cli-stats.json')
+  var stats
+  try {
+    stats = JSON.parse(fs.readFileSync(statsFile))
```

Comment:
Is it premature to abstract the metrics out into a defined interface, if our goal is to introduce more supportability metrics over time? In other words, would it make sense to put the metrics object into a module that can be required as needed across the project (or maybe hung off a `process.on` listener, a la numbat)?

I'm fine with shipping this as-is, but it does seem like this is something we're going to want to change pretty soon, and having it behind a defined interface from the get-go therefore makes a certain amount of sense.